modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/types/PartialUpdate.scala

```diff
@@ -0,0 +1,39 @@
+package ch.epfl.bluebrain.nexus.iam.service.types
+
+import ch.epfl.bluebrain.nexus.commons.iam.acls.Permissions
+import ch.epfl.bluebrain.nexus.commons.iam.identity.Identity
+import io.circe.generic.extras.Configuration
+import io.circe.{Decoder, Encoder, Json}
+
+/**
+  * Base enumeration type for PartialUpdate message classes.
+  */
+sealed trait PartialUpdate extends Product with Serializable {
+
+  /**
+    * @return the context object of the partial update message
+    */
+  def `@context`: Option[Json]
```

Comment:
I don't think we should have an `@context` in the code representation, but we should handle it in the marshalling and unmarshalling.